src/bindings.rs

```diff
@@ -643,32 +644,173 @@ publish = false
     }
 }
 
-fn export_world_key_path(resolve: &Resolve, key: &WorldKey) -> String {
-    match key {
-        WorldKey::Name(name) => format!("bindings::exports::{name}", name = name.to_snake_case(),),
-        WorldKey::Interface(id) => {
-            let iface = &resolve.interfaces[*id];
-            let pkg = &resolve.packages[iface.package.expect("interface should have a package")];
-            let name = iface.name.as_deref().expect("unnamed interface");
-            format!(
-                "bindings::exports::{ns}::{pkg}::{name}",
-                ns = pkg.name.namespace.to_snake_case(),
-                pkg = pkg.name.name.to_snake_case(),
-                name = name.to_snake_case(),
+/// Represents a node in a "use" trie.
+#[derive(Default)]
+struct UseTrieNode {
+    // Map of segment index to trie node
+    children: HashMap<usize, UseTrieNode>,
+    // Set of type indexes that are used at this node
+    tys: HashSet<usize>,
```

Comment:
Since the iteration order of `tys` determines what order the types are printed in, could these use `indexmap::IndexMap` and `indexmap::IndexSet` here instead, so that the output is deterministic?

Reply:
Originally I left the order unstable as `rustfmt` would sort them, but this should be fixed if we don't want to rely on `rustfmt` with similar justification as to why use a trie in the first place.

I'll fix this.